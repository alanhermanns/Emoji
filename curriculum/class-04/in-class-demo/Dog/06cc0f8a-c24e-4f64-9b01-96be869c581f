{"name":"spot","age":5,"weight":"20 lbs","id":"06cc0f8a-c24e-4f64-9b01-96be869c581f"}