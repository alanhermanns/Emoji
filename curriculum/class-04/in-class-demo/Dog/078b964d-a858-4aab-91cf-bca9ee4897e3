{"name":"spot","age":5,"weight":"20 lbs","id":"078b964d-a858-4aab-91cf-bca9ee4897e3"}