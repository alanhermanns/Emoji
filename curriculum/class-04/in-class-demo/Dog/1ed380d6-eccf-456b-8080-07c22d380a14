{"name":"spot","age":5,"weight":"20 lbs","id":"1ed380d6-eccf-456b-8080-07c22d380a14"}