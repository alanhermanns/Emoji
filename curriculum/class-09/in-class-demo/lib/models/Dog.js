const mongoose = require('mongoose');

const schema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  dob: {
    type: Date,
    required: true
  },
  weight: {
    type: String,
    required: true
  }
});

schema.virtual('age')
  .get(function() {
    const now = new Date();
    return now.getFullYear() - this.dob.getFullYear();
  })
  .set(function(val) {
    const currentBirthYear = this.dob.getFullYear();
    const diff = val - this.age;
    this.dob.setYear(currentBirthYear - diff);
  });

module.exports = mongoose.model('Dog', schema);
