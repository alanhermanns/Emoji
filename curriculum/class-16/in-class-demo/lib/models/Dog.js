const mongoose = require('mongoose');

const schema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  age: {
    type: Number,
    required: true,
    min: 0
  },
  weight: String,
  settings: Object
}, {
  toJSON: {
    virtuals: true
  }
});

// get -> Dog.findById('someID').then(myDog => myDog.firstLetter);
// set -> Dog.findById('someID').then(myDog => myDog.firstLetter = 'b');
schema.virtual('firstLetter')
  .get(function() {
    return this.name[0];
  })
  .set(function(newFirstLetter) {
    this.name = `${newFirstLetter}${this.name.slice(1)}`;
  });

// $lookup
// Dog.findById('someID').then(myDog => myDog.owners);
schema.virtual('owner', {
  ref: 'Person',
  localField: '_id',
  foreignField: 'dogId'
});

// Dog.findByName('spot').then(allSpots => console.log(allSpots));
schema.statics.findByName = function(name) {
  return this
    .find({ name });
};

// Dog.findById('someID').then(myDog => myDog.findSimilar());
schema.methods.findSimilar = function() {
  const currentDog = this;
  return this
    .model()
    .find({ name: currentDog.name, age: currentDog.age });
};

// collection -> dogs
module.exports = mongoose.model('Dog', schema);
