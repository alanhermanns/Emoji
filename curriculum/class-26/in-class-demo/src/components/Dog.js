import React, { Component } from 'react';
import PropTypes from 'prop-types'
import Health from './Health';
import DataListElement from './DataListElement';

export default class Dog extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    weight: PropTypes.string.isRequired
  }

  render() {
    const { name, age, weight } = this.props;
    return (
      <>
        <Health />
        <dl>
          <DataListElement
            title="Name"
            description={name} />
          <DataListElement
            title="Age"
            description={age} />
          <DataListElement
            title="Weight"
            description={weight} />
        </dl>
      </>
    )
  }
}
