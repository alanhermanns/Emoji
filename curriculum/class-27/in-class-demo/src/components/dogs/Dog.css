.Dog {
  dt {
    color: red;
  }

  dd {
    color: yellow;
  }
}
