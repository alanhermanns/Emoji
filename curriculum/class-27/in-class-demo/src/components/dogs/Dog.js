import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Dog.css';

export default class Dog extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    weight: PropTypes.string
  }

  speak = () => {
    console.log('woof');
  }

  render() {
    const { name, age, weight } = this.props;

    return (
      <dl className={styles.Dog}>
        <dt>Name</dt>
        <dd>{name}</dd>

        <dt>Age</dt>
        <dd>{age}</dd>

        {weight ? (
          <>
            <dt>Weight</dt>
            <dd>{weight}</dd>
          </>
        ) : <p>No Weight sorry</p>}
        <button onClick={this.speak}>Speak</button>
      </dl>
    );
  }
}
