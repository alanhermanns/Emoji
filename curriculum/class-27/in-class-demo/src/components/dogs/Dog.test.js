import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('Dog component', () => {
  it('renders Dog', () => {
    const wrapper = shallow(<Dog name="Spot" age={5} weight="20 lbs"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
