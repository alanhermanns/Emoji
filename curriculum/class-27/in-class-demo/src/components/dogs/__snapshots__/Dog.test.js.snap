// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders Dog 1`] = `
<dl
  className="Dog"
>
  <dt>
    Name
  </dt>
  <dd>
    Spot
  </dd>
  <dt>
    Age
  </dt>
  <dd>
    5
  </dd>
  <dt>
    Weight
  </dt>
  <dd>
    20 lbs
  </dd>
</dl>
`;
